egs/librispeech/ASR/pruned_transducer_stateless2/scaling.py

```diff
@@ -111,6 +111,74 @@ def backward(
         return x_grad - neg_delta_grad, None, None, None, None, None, None
 
 
+class GradientFilterFunction(torch.autograd.Function):
+    @staticmethod
+    def forward(
+        ctx,
+        x: Tensor,
+        batch_dim: int,  # e.g., 1
+        threshold: float,  # e.g., 10.0
+        *params: nn.Parameter,  # module parameters
```

Comment:
better to say Tensor here, because the scaled parameters are actually Tensor not Parameter, and anyway Parameter is a subclass of Tensor

Reply:
Also, `params` is of type `tuple`.